### ISSUE TITLE
Embedding scaling is tied to positional encoding for Transformer models

### ISSUE THREAD

**Issue by guillaumekln** [OpenNMT/OpenNMT-py#1722]
When positional encoding is disabled, the embedding scaling is also disabled even though the operations are independent:

https://github.com/OpenNMT/OpenNMT-py/blob/1.0.0/onmt/modules/embeddings.py#L48

In consequence, Transformer models with relative position representations do not follow the reference implementation which scales the embedding [by default](https://github.com/tensorflow/tensor2tensor/blob/e1f0e3a746bb322f4bf3975fad2c8105b3a43a49/tensor2tensor/layers/common_hparams.py#L111).

**Comment by francoishernandez** [comment]
Nice catch.
Do you think this scaling factor makes much difference when training from scratch?
I agree it would be better to stick to the original implementation though.

**Comment by guillaumekln** [comment]
I don't think this can make an interesting difference, especially as the next operation is the layer normalization. So it's mostly about clarity.

**Comment by vince62s** [comment]
I really think it is not necessary. the scaling of embeddings is necessary in the case of Sinusoidal position encoding to make the addition of two elements of the same order range.
However there is an issue with dropout which is applied only with PE. It should be applied in even without. #2285
